src/index.ts

```diff
@@ -57,7 +57,7 @@ export interface Context {
   moduleDirectory: string[];
 }
 
-async function main(args: CliArguments) {
+export async function main(args: CliArguments) {
```

Comment:
I dislike exporting things just for tests, but comments above explain why.  I did contemplate moving `main` out in `index.ts` into it's own file as I _think_ the `require` in the test is still going to run `yargs` and everything even though we probably don't want it to.